packages/playwright-core/src/utils/isomorphic/urlMatch.ts

```diff
@@ -107,19 +107,15 @@ export function urlMatches(baseURL: string | undefined, urlString: string, match
     match = globToRegex(match);
   if (isRegExp(match))
     return match.test(urlString);
-  if (typeof match === 'string' && match === urlString)
-    return true;
-  const url = parsedURL(urlString);
+  const url = parseURL(urlString);
```

Comment:
This seemed to be a "fast path" but it was never executed since if its a string, its always a regex afterwards.